### PR TITLE
Created basic database model

### PR DESCRIPTION
**Pull request by MatusHromulak**
I have created a basic database model that should fit all requirements. There two open questions for now:
1. How many campaigns do we want to have running per basket? Currently set to just one.
2. User authentication implementation. For now the hash and salt are stored in the user table.

@all, please review and post suggestions and comments.

### CODE REVIEW

**Review by MartinCernohorsky — COMMENTED** [review]
Hi, good job indeed. Since this is something completely out of  my league, please take this just as a few question to help me understand what is it doing.

BasketItem.Count - can be changed to quantity/qty/qt ? Just for the sake of clarity.

Order.BillingInformation is mandatory therefore payments could be made only by credit card?

BaskedID and UserID meets each other when the order is created. How do pair them together before that? Cookies?

Campaign - is it possible store a time period (or status) when the campaign is valid? 

Product/Campaign -just to make final calculations for basket more fun, is it possible to create table ProductType, assign each product a type and do campaigns aimed for certain product type?

Storage - is it worth the effort to add storage name in human readable form? The only thing that can distinguish two storage rooms at the same address is DoorNumber.

I don't see any reason not to use this model.

File: ServiceBus/ServiceBus/Basket.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            this.BasketItems = new HashSet<BasketItem>();
+        }
+    
+        public int Id { get; set; }
+        public Nullable<int> CampaignId { get; set; }
+        public double TotalPrice { get; set; }
+    
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
+        public virtual ICollection<BasketItem> BasketItems { get; set; }
+        public virtual Campaign Campaign { get; set; }
```
Comment (mozdren) [inline ServiceBus/ServiceBus/Basket.cs]: We should be able to use multiple campaigns in one basket (discounts could be combined - that is usually called combideal in e-comerce world).

**Review by mozdren — CHANGES_REQUESTED** [review]
I see you have made a really good progress in the database model, but in some cases in seams not complete.

Please, check the data contracts that were implemented and compare it to yours db model. Then, contact me and we can discuss what could be changed exactly (in model or data contracts).

Main notes:

Since the orders are being formed during customers visit on web page, we should be able to fill address a bit later that immediately when the entity is being created. That is also true for billing information etc. . Please consider changes to your model or current data contracts. This is quite a hard task and will influence direction of next development. Thank you.

**Comment by MatusHromulak** [comment]
Hello, I greatly appreciate the comments.

_BasketItem.Count - can be changed to quantity/qty/qt ? Just for the sake of clarity._
Yes, that will make it more clear. Will implement.

_Order.BillingInformation is mandatory therefore payments could be made only by credit card?_
I need to rethink that part, considering multiple payment options.

_BaskedID and UserID meets each other when the order is created. How do pair them together before that? Cookies?_
Honestly, no idea... this needs to be brainstormed.

_Campaign - is it possible store a time period (or status) when the campaign is valid?_
The campaign needs to be expanded a little, will do.

_Product/Campaign -just to make final calculations for basket more fun, is it possible to create table ProductType, assign each product a type and do campaigns aimed for certain product type?_
Makes good sense. Will implement.

_Storage - is it worth the effort to add storage name in human readable form? The only thing that can distinguish two storage rooms at the same address is DoorNumber._
Same as above.

_Since the orders are being formed during customers visit on web page, we should be able to fill address a bit later that immediately when the entity is being created. That is also true for billing information etc. ._
It is true that I was thinking in more of an DB admin way, compared to a user-experience based approach. Will rework.

I will review the model and the implement changes some time by the end of this week. Further comments are welcomed. Cheers.

**Comment by MatusHromulak** [comment]
I updated the model to some extent. My remaining problem is the BillingInforamtion table. I modeled it based on the existing data contracts. But what it the purpose? Are that the billing information of the company, where the user is expected to pay? Then it should be probably done differently. Or are those supposed to be the information of the user? There is no point in storing their bank details...

**Comment by mozdren** [comment]
Billing information is not mandatory, but useful for the future implementation, since we might use this information when considering real registration of a customer. Also it is not mandatory to have this information available right away.

**Review by mozdren — APPROVED** [review]

**Pull request merged**